### PR TITLE
feat: Auto cabin ready

### PR DESCRIPTION
**Pull request by tracernz**
## Summary of Changes
Request from Boris.

- During departure, the cabin ready memo is auto-triggered 2-3 minutes after flight phase 2 (first engine started) is entered.
- During arrival, the cabin ready memo is auto-triggered 2-8 seconds after the gear is extended.

## Additional context

I have split #5683 into it's own PR.

Discord username (if different from GitHub):

## Testing instructions

Prepare for some takeoffs and ensure the cabin announcement is made automatically 2-3 minutes after the first engine is started. Prepare for some landings and ensure the cabin announcement happens a short time after extending the gear. Make sure the annoucements don't happen at other stages during the flight. Check that the overhead call buttons can still trigger the cabin ready.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by FlyingCarpet7** [comment]
A quick test now and it seems to work as it should! 

Regarding the GND CTL button. I am not sure if it can be pushed ON when it’s already on in auto. Probably not so leave it like this until further proof.

Tried a go around and quick circuit back to land and the cabin ready event was retriggered. Probably the cabin ready switch would have been left in ready position by the flight attendant duiring such event. Maybe the ready event logic could maintain ready and only reset itself  when the seatbelt switch has been switched off once?

**Comment by komp1821** [comment]
> Regarding the GND CTL button. I am not sure if it can be pushed ON when it’s already on in auto. Probably not so leave it like this until further proof.

Took a look at the schematics. Can confirm you're right. When the auto condition is fulfilled, the ON light cannot come on if the pushbutton is pushed again.

I would like to add that "nose landing gear extended" is another auto condition. While it's rather unlikely that you will lift off the ground without engines, it could happen to you that both engines fail in flight. In this case, the recorder still keeps recording.

**Comment by MisterChocker** [comment]
@tracernz Is your implementation following @FlyingCarpet7 and @fromm1990 suggestion/observation that the GND CTL button cant be enabled when in auto mode?

**Comment by tracernz** [comment]
> @tracernz Is your implementation following @FlyingCarpet7 and @fromm1990 suggestion/observation that the GND CTL button cant be enabled when in auto mode?

It will momentarily enable and then disable if an engine is running. To make it not even momentarily enable will require some more XML changes... I might split this PR in two. Initially I did them together because they were requested at the same time by FlyingCarpet and Boris, and there would be merge conflicts if they were separate but it's clear now they are not going to be quick merges so that doesn't matter anymore.

**Comment by FlyingCarpet7** [comment]
I think we should merge this one.  If anyone want to fine tune it later it can be open again.

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml [hunk 1]
```diff
@@ -2701,10 +2701,11 @@
...
                         <Component ID="RCDR">
-                            <UseTemplate Name="FBW_Push_Toggle">
+                            <UseTemplate Name="FBW_Push_Held">
                                 <NODE_ID>PUSH_OVHD_RCDR_GNDCTL</NODE_ID>
                                 <PART_ID>PUSH_OVHD_RCDR_GNDCTL</PART_ID>
-                                <TOGGLE_SIMVAR>L:A32NX_RCDR_GROUND_CONTROL_ON</TOGGLE_SIMVAR>
+                                <LEFT_SINGLE_CODE>(L:A32NX_RCDR_GROUND_CONTROL_ON) ! (&gt;L:A32NX_RCDR_GROUND_CONTROL_ON)</LEFT_SINGLE_CODE>
+                                <NO_LEFT_LEAVE_CODE>True</NO_LEFT_LEAVE_CODE>
                                 <SEQ_POWERED>(L:A32NX_ELEC_AC_1_BUS_IS_POWERED, Bool)</SEQ_POWERED>
                                 <SEQ2_CODE>(L:A32NX_RCDR_GROUND_CONTROL_ON, Bool)</SEQ2_CODE>
```
Comment (MisterChocker) [inline flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:2710]: I believe this controls the ON light.

Use the AUTO local variable as well to only show the ON light when AUTO is false.

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js [hunk 2]
```diff
@@ -83,7 +84,19 @@ class A32NX_LocalVarUpdater {
                 type: "Bool",
                 selector: this._areSlidesArmed.bind(this),
                 refreshInterval: 100,
-            }
+            },
+            {
+                varName: "L:A32NX_RCDR_GROUND_CONTROL_ON",
```
Comment (MisterChocker) [inline flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js:89]: name this to auto

File: flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml [hunk 3]
```diff
@@ -2701,10 +2701,11 @@
                         </Component>
 
                         <Component ID="RCDR">
-                            <UseTemplate Name="FBW_Push_Toggle">
+                            <UseTemplate Name="FBW_Push_Held">
                                 <NODE_ID>PUSH_OVHD_RCDR_GNDCTL</NODE_ID>
                                 <PART_ID>PUSH_OVHD_RCDR_GNDCTL</PART_ID>
-                                <TOGGLE_SIMVAR>L:A32NX_RCDR_GROUND_CONTROL_ON</TOGGLE_SIMVAR>
+                                <LEFT_SINGLE_CODE>(L:A32NX_RCDR_GROUND_CONTROL_ON) ! (&gt;L:A32NX_RCDR_GROUND_CONTROL_ON)</LEFT_SINGLE_CODE>
```
Comment (MisterChocker) [inline flybywire-aircraft-a320-neo/SimObjects/AirPlanes/FlyByWire_A320_NEO/model/A320_NEO_INTERIOR.xml:2707]: also only allow ON local var to be set to true when AUTO is false

**Review by MisterChocker — DISMISSED** [review]

**Comment by FlyingCarpet7** [comment]
Can we merge this one pleas? I am happy with the current functionality.

File: flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js [hunk 4]
```diff
@@ -211,5 +224,19 @@ class A32NX_LocalVarUpdater {
         );
     }
 
+    cvrGroundControl(deltaTime, identifier, currentValue) {
+        if (currentValue) {
+            const isEngineOneRunning = SimVar.GetSimVarValue("ENG COMBUSTION:1", "bool");
+            const isEngineTwoRunning = SimVar.GetSimVarValue("ENG COMBUSTION:2", "bool");
+            return !isEngineOneRunning && !isEngineTwoRunning;
+        }
+        return currentValue;
```
[the remark this answers is not in the document]
Comment (tracernz) [reply flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_LocalVarUpdater.js:233]: This has nothing to do with the FMGC. It is purely an electromechanical circuit with external signals from an LGCIU (grd/flt) and the engines (off or on).

**Review by 2hwk — APPROVED** [review]

**Review by MisterChocker — APPROVED** [review]

**Comment by John-UK** [comment]
I spawned at a stand in EGLL starting the APU and remaining at the stand. When starting the first engine the cabin ready memo was displayed inconsistently.  See attached. Each time the cabin memo  was seen it was written at the same time as pred W/S off and an audible tone is heard. I could force the cabin memo with a call button.
Cabin1
Cabin2

**Pull request closed without merging**